certbot-apache/certbot_apache/augeasparser.py

```diff
@@ -146,12 +148,26 @@ def add_child_block(self, name, parameters=None, position=None):  # pragma: no c
     # pylint: disable=unused-argument
     def add_child_directive(self, name, parameters=None, position=None):  # pragma: no cover
         """Adds a new DirectiveNode to the sequence of children"""
-        new_metadata = {"augeasparser": self.parser, "augeaspath": assertions.PASS}
-        new_dir = AugeasDirectiveNode(name=assertions.PASS,
-                                      ancestor=self,
-                                      filepath=assertions.PASS,
+
+        if not parameters:
+            raise errors.PluginError("Directive requires parameters and none were set.")
+
+        insertpath, realpath, before = self._aug_resolve_child_position(
+            "directive",
+            position
+        )
+        new_metadata = {"augeasparser": self.parser, "augeaspath": realpath}
+
+        # Create the new directive
+        self.parser.aug.insert(insertpath, "directive", before)
+        # Set the directive key
+        self.parser.aug.set(realpath, name)
```

Comment:
what's this? why didn't we need to do it with `add_child_block`?

Reply:
That's because blocks use their name as a key in Augeas XPATH notation, while directives always use key "directive", we need to set the name for it.